File: scripts/cross_lang_test.js
// Cross-language test: Load Python-trained model in JS
const { loadModel } = require('train-lle.js');

async function test() {
  const model = await loadModel('model_py.lle');
  const input = [1, 2, 3, 4]; // example
  const pred = model.forward(input);
  console.log('Prediction:', pred);
  // Verify against expected
  const expected = [0.5, 0.6]; // placeholder
  if (Math.abs(pred[0] - expected[0]) < 0.01) {
    console.log('Test passed');
  } else {
    console.log('Test failed');
  }
}

test();

File: archive/cross_lang_test.py
import train_lle

def test():
    model = train_lle.load_model('model_js.lle')
    input_data = [1, 2, 3, 4]
    pred = model.predict(input_data)
    print('Prediction:', pred.data)
    expected = [0.5, 0.6]
    if abs(pred.data[0] - expected[0]) < 0.01:
        print('Test passed')
    else:
        print('Test failed')

if __name__ == '__main__':
    test()